test_conformance/commonfns/test_unary_fn.cpp

```diff
@@ -0,0 +1,379 @@
+}
+
+//--------------------------------------------------------------------------
+template <typename T>
+int verify_degrees(const T *const inptr, const T *const outptr, int n)
+{
+    float error, max_error = 0.0f;
+    double r, max_val = NAN;
+    int max_index = 0;
+
+    for (int i = 0, j = 0; i < n; i++, j++)
+    {
+        r = (180.0 / M_PI) * inptr[i];
+
+        error = UlpFn(outptr[i], r);
+
+        if (fabsf(error) > max_error)
+        {
+            max_error = error;
+            max_index = i;
+            max_val = r;
+            if (fabsf(error) > MAX_ERR)
+            {
+                log_error("%d) Error @ %a: *%a vs %a  (*%g vs %g) ulps: %f\n",
+                          i, inptr[i], r, outptr[i], r, outptr[i], error);
+                return 1;
+            }
+        }
+    }
+
+    log_info("degrees: Max error %f ulps at %d: *%a vs %a  (*%g vs %g)\n",
+             max_error, max_index, max_val, outptr[max_index], max_val,
+             outptr[max_index]);
+
+    return 0;
+}
+
+//--------------------------------------------------------------------------
+template <typename T>
+int verify_radians(const T *const inptr, const T *const outptr, int n)
+{
+    float error, max_error = 0.0f;
+    double r, max_val = NAN;
+    int max_index = 0;
+
+    for (int i = 0, j = 0; i < n; i++, j++)
+    {
+        r = (M_PI / 180.0) * inptr[i];
+        error = Ulp_Error(outptr[i], r);
+        if (fabsf(error) > max_error)
+        {
+            max_error = error;
+            max_index = i;
+            max_val = r;
+            if (fabsf(error) > MAX_ERR)
+            {
+                log_error("%d) Error @ %a: *%a vs %a  (*%g vs %g) ulps: %f\n",
+                          i, inptr[i], r, outptr[i], r, outptr[i], error);
+                return 1;
+            }
+        }
+    }
+
+    log_info("radians: Max error %f ulps at %d: *%a vs %a  (*%g vs %g)\n",
+             max_error, max_index, max_val, outptr[max_index], max_val,
+             outptr[max_index]);
+
+    return 0;
+}
+
+//--------------------------------------------------------------------------
+template <typename T>
+int verify_sign(const T *const inptr, const T *const outptr, int n)
+{
+    T r = 0;
+    for (int i = 0; i < n; i++)
+    {
+        if (inptr[i] > 0.0f)
+            r = 1.0;
+        else if (inptr[i] < 0.0f)
+            r = -1.0;
+        else
+            r = 0.0;
+        if (r != outptr[i]) return -1;
+    }
+    return 0;
+}
+
+}
+
+//--------------------------------------------------------------------------
+
+template <typename T>
+int test_unary_fn(cl_device_id device, cl_context context,
+                  cl_command_queue queue, int n_elems,
+                  const std::string &fnName, VerifyFuncUnary<T> verifyFn)
+{
+    clMemWrapper streams[2];
+    std::vector<T> input_ptr, output_ptr;
+
+    std::vector<clProgramWrapper> programs;
+    std::vector<clKernelWrapper> kernels;
+
+    int err, i;
+    MTdata d;
```

Comment:
A modernization here would be to use `MTdataHolder` instead, but not sure how far you want to go with changing the existing code.

Reply:
Thanks, I replaced `MTdata` with the wrapper in `commonfns` source code files